src/Services/CreditCards/ICreditCardsServices.cs

```diff
@@ -0,0 +1,151 @@
+            if (CardToDelete != null)
+            {
+                _appDbContext.UserCreditCards.Remove(CardToDelete);
+                await _appDbContext.SaveChangesAsync();
+                return new Result
+                {
+                    IsSuccssfull = true,
+                    Message = "successfuly removed credit card # "+id+" from database"
+                };
+            }
+            else
+            {
+                return new Result
+                {
+                    IsSuccssfull = false,
+                    Message = "card not found in database"
+                };
+            }
+            
+            
+        }
+        /*
+         Gets Credit Card for one user by the Cards Id 
+         */
+        public async Task<Result<CreditCardsDto>> GetUserCreditCardAsync(int id)
+        {
+            var userCredit = await _appDbContext.UserCreditCards.FindAsync(id);
+            if (userCredit == null)
+            {
+                return new Result<CreditCardsDto> { IsSuccssfull = false, Message = "No Credit Card found !" };
+            }
+            else
+            {
+                return new Result<CreditCardsDto>
+                {
+                    IsSuccssfull = true,
+                    Data = new CreditCardsDto
+                    {
+                        CardSecretCode = userCredit.CreditCardSecretCode,
+                        CreditCardNumber = userCredit.CreditCardNumber,
+                        ExpireMonth = userCredit.ExpireMonth,
+                        ExpireYear = userCredit.ExpireYear,
+                        HolderName = userCredit.HolderName
+                    }
+                };
+            }
+        }
+        /*
+         Gets All the Credit Cards for one user
+         */
+        public async Task< Result<List<CreditCardsDto>> > GetUserCreditCardsForUserAsync(int id)
+        {
+            List<UserCreditCard> cards = await _appDbContext.UserCreditCards.Where(x => x.UserId == id).ToListAsync();
+            if(cards.Count == 0 | cards == null)
+            {
+                return new Result<List<CreditCardsDto>>
+                {
+                    IsSuccssfull = false,
+                    Message = "User Doesn't Have any cards !"
+                };
+            }
+            List<CreditCardsDto> cardsDtos = new List<CreditCardsDto>();
+            foreach(var Card in cards)
+            {
+                
+                cardsDtos.Add(new CreditCardsDto().ToCreditCardsDto(Card));
+            }
+            return new Result<List< CreditCardsDto>> { IsSuccssfull = true, Data = cardsDtos };
+        }
+
+        public async Task<Result<CreditCardsDto>> UpdateUserCreditCardAsync(int id, UserCreditCard userCreditCard)
+        {
+            try
+            {
+                var card = await _appDbContext.UserCreditCards.SingleOrDefaultAsync(x => x.Id == id);
```

Comment:
Please use find to find item by its Id, Single still iterates over the entire list to make sure there is a single item in the list regardless if it was found (we dont want that behavior as PK is already unique )